src/xref2/component.ml

```diff
@@ -455,6 +455,12 @@ and CComment : sig
 end =
   CComment
 
+and Label : sig
+  type t =
+    [ `Heading of Odoc_model.Comment.heading ] Odoc_model.Comment.with_location
```

Comment:
I'm not convinced by this. In all of the other Components we use `Ident.t` rather than identifier (e.g. in the above definition of `CComment`). This is because we use these components for doing expansions and often end up placing them in different contexts than they were originally found - this usually works because we end up turning the local `Ident.t` back into a global `Identifier.t` at the right place, but if you're preserving the `Identifier` it'll be incorrect. I think what you actually want is to pick out the `Heading` from the `CComment` module above as a separate type and use that instead.

Reply:
This is fixed. I've also made `Label.t` a record to make sure we catch every places where the conversion is needed.